src/engine.cc

```diff
@@ -0,0 +1,88 @@
+#include "engine.h"
+#include "gobject.h"
+
+#include <eos-knowledge-content.h>
+
+using namespace v8;
+
+struct CallbackData {
+  Persistent<Function> resolve;
+  Persistent<Function> reject;
+};
+
+static void EngineGetObjectCallback(GObject *source,
+                                    GAsyncResult *result,
+                                    gpointer user_data)
+{
+  Isolate *isolate = Isolate::GetCurrent ();
+  CallbackData *data = (CallbackData *)user_data;
+
+  GError *error = NULL;
+  EkncContentObjectModel *model;
+
+  if ((model = eknc_engine_get_object_finish (EKNC_ENGINE (source), result, &error))) {
+    const unsigned argc = 1;
+    Local<Value> argv[argc] = { WrapperFromGObject (isolate, G_OBJECT (model)) };
+    Local<Function>::New(isolate, data->resolve)->Call(isolate->GetCurrentContext()->Global(), argc, argv);
```

Comment:
Not sure, I think it might. But the functions here are internal to the index.js and not exposed, so I don't think we need to worry about use cases besides the one we have directly.